### PR TITLE
Fix: new freelancer registration

### PR DESCRIPTION
**Pull request by cimafrancamae**
## Changelog

- added `rescue ActiveRecord::RecordInvalid` on `create_freelancer`
- refactored `create_freelancer` logic
- specified `skills` type and added `role_id` in params
- refactored `checkbox` field in form

### Explanation

The reason why the `role` was not updating was there were invalid values in `sign_up_params`. The controller had no `rescue` method, and the server automatically `rollback` the transaction.

### CODE REVIEW

File: app/javascript/controllers/ui/checkbox_controller.js
Comment (TyJacalan) [inline app/javascript/controllers/ui/checkbox_controller.js:1]: forgot to remove logger

File: app/views/devise/registrations/new_freelancer.html.erb [hunk 1]
```diff
@@ -1,52 +1,47 @@
...
   <div class="field">
-    <%= f.label :mobile %><br />
-    <%= f.text_field :mobile %>
-    <%= f.error_message :mobile %>
+    <%= f.label :skills %>
+    <div class="skills-checkboxes">
+      <% @categories.each do |category| %>
+        <div>
+          <%= check_box_tag "user[skills][]", category.title, @user.skills.include?(category.title.to_s), class: "peer h-4 w-4 shrink-0 rounded-sm border border-primary hover:cursor-pointer focus-visible:outline-none focus-visible:ring-0 disabled:cursor-not-allowed disabled:opacity-50" %>
+          <%= label_tag "user_skills_#{category.title}", category.title %>
```
Comment (TyJacalan) [inline app/views/devise/registrations/new_freelancer.html.erb:36]: this also works pala
```
<% @categories.each do |category| %>
   <div class='flex flex-row gap-2'>
      <%= f.checkbox category.title %>
      <%= f.label category.title %>
   </div>
<% end %>
```

**Review by TyJacalan — APPROVED** [review]
Nice fix! Would have never caught that!

**Review by p3rc1us — APPROVED** [review]

File: app/views/devise/registrations/new_freelancer.html.erb (hunk 1 above)
Comment (cimafrancamae) [reply app/views/devise/registrations/new_freelancer.html.erb:36]: This doesn't pass the values of the selected categories 🤔

**Pull request merged**